core/models.py

```diff
@@ -70,6 +81,45 @@ def validate_time(time, field_name):
             {field_name: _("Date/time can not be in the future.")}, code="time_invalid"
         )
 
+TAG_COLORS = [
+    "#FF0000", "#00FF00", "#0000FF", "#FF00FF", "#FFFF00", "#00FFFF",
+    "#FF7F7F", "#7FFF7F", "#7F7FFF", "#FF7FFF", "#FFFF7F", "#7FFFFF",
+    "#7F0000", "#007F00", "#00007F", "#7F007F", "#7F7F00", "#007F7F",
+]
+
+def random_color():
+    return TAG_COLORS[random.randrange(0, len(TAG_COLORS))]
+
+class BabyBuddyTag(TagBase):
+    class Meta:
+        verbose_name = _("Tag")
+        verbose_name_plural = _("Tags")
+    
+    color = models.CharField(
+        "Color",
+        max_length=32,
+        default=random_color,
+        validators=[RegexValidator(r"^#[0-9A-F]{6}$")]
+    )
+
+    last_used = models.DateTimeField(
+        default=now,
+        blank=False,
+    )
+
+
+class BabyBuddyTagged(GenericTaggedItemBase):
```

Comment:
Ditto here re: the prefix -- any reason we can't just use `Tagged` for the model name?